python/ray/serve/_private/proxy_state.py

```diff
@@ -46,39 +39,40 @@ def actor_id(self) -> str:
         raise NotImplementedError
 
     @abstractmethod
-    def start_new_ready_check(self):
-        """Start a new ready check on the proxy actor."""
-        raise NotImplementedError
-
-    @abstractmethod
-    def start_new_health_check(self):
-        """Start a new health check on the proxy actor."""
-        raise NotImplementedError
-
-    @abstractmethod
-    def start_new_drained_check(self):
-        """Start a new drained check on the proxy actor.
+    def is_ready(self, timeout_s: float) -> Optional[bool]:
+        """Return the payload from proxy ready check when ready.
 
-        This is triggered once the proxy actor is set to draining. We will leave some
-        time padding for the proxy actor to finish the ongoing requests. Once all
-        ongoing requests are finished and the minimum draining time is met, the proxy
-        actor will be transition to drained state and ready to be killed.
+        Since actual readiness check is asynchronous, this method could return
+        either of the following statuses:
+            - None: Readiness check is pending
+            - True: Readiness check completed successfully (proxy is ready)
+            - False: Readiness check completed with failure (either timing out
+            or failing)
```

Comment:
nit: `None`/`True`/`False` behavior split is a little implicit, consider an enum